### ISSUE TITLE
custom models not included in saved projects (Trac #1016)

### ISSUE THREAD

**Issue by pkienzle** [SasView/sasview#1085]
Migrated from http://trac.sasview.org/ticket/1016
```json
{
    "status": "new",
    "changetime": "2019-03-03T14:58:00",
    "_ts": "2019-03-03 14:58:00.318105+00:00",
    "description": "Need to include code for custom models in saved projects in order to share them with others.  In particular, when receiving a saved project in a bug report, it would be very useful if said project contained the custom model.",
    "reporter": "pkienzle",
    "cc": "",
    "resolution": "",
    "workpackage": "SasView Bug Fixing",
    "time": "2017-10-24T20:26:11",
    "component": "SasView",
    "summary": "custom models not included in saved projects",
    "priority": "major",
    "keywords": "",
    "milestone": "SasView 5.0.0",
    "owner": "",
    "type": "defect"
}
```

**Comment by rozyczko** [comment]
Trac update at `2019/01/22 10:56:09`: **piotr** commented:

This is working as expected in the ESS_GUI branch.

**Comment by butlerpd** [comment]
Trac update at `2019/03/03 14:58:00`: **butler** commented:

So is this a "fix" or does someone need to verify that @piotr is understanding the same thing as @pkienzle before we can close this ticket?

**Comment by rozyczko** [comment]
Right. I conveniently missed the "include code for custom models in saved projects" part here.
This still does need addressing.

**Comment by butlerpd** [comment]
Hummm.... Is this going to be too much for 5.0 at this point?  It is a bit of a problem for save projects but then again save projects has always worked rather poorly in the past? (sort of "it is ok to be bad if the old was bad" :-)

**Comment by pkienzle** [comment]
There's a security issue here: by loading a saved project you are executing arbitrary compiled C code on your computer. Maybe run a quick parse on it through a restricted C language to check that it is only accessing a restricted set of functions.  Anything with pointer arithmetic or array indexing will need to be examined by hand.  Keeping track of what has been approved and what hasn't is an additional problem; don't want to be asking every time the file is loaded.

So yes: too much for 5.0.
